src/dealer_router.rs

```diff
@@ -91,20 +92,26 @@ impl Socket for RouterSocket {
                 peers: Arc::new(DashMap::new()),
             }),
             _accept_close_handle: None,
+            binds: Vec::new(),
         }
     }
 
     async fn bind(&mut self, endpoint: impl TryIntoEndpoint + 'async_trait) -> ZmqResult<()> {
-        let endpoint = endpoint.try_into()?;
+        let mut endpoint = endpoint.try_into()?;
         let stop_handle =
-            util::start_accepting_connections(&endpoint, self.backend.clone()).await?;
+            util::start_accepting_connections(&mut endpoint, self.backend.clone()).await?;
```

Comment:
I would consider changing the API the following way:
```
async fn bind(&mut self, endpoint: &str) -> ZmqResult<Endpoint>
```
and parse endpoint further down in utils method. making it a return value of the `start_accepting_connections` call